src/data/real/ennreal.lean

```diff
@@ -1371,6 +1371,42 @@ begin
   exact n.lt_two_pow
 end
 
+lemma Union_Ioi_add_inv_nat (z : ℝ≥0∞) : (⋃ (n : ℕ), Ioi (z + n⁻¹)) = Ioi z :=
```

Comment:
I added the lemma you suggested, `Union_Ioi_eq_Ioi_infi`. But it is not clear it leads to a simplification here. I rewrote the proof (and also tested directly for the proof of the next lemma, if the current one is too specialized), and it is almost the same length and I think uses less familiar ideas.

I of course agree that `Union_Ioi_eq_Ioi_infi` is in principle good to have, and I'll try to find the right place for it, but I would consider reverting the proofs here to the original ones, or similar, with only the ones that are actually used in this PR.

Reply:
There is [`ennreal.add_infi`](https://leanprover-community.github.io/mathlib_docs/data/real/ennreal.html#ennreal.add_infi), and it would be nice if you could add lemmas computing `⨅ i, (f i)⁻¹` (I think that is `(⨆ i, f i)⁻¹` assuming `f i ≠ 0`).

But as I said, I'm not sure if this special case is worth having, especially since you're not using it.